From 67e7023d34fbd425932a9f53cc50665b9de9916f Mon Sep 17 00:00:00 2001
From: Piotr Gorski <lucjan.lucjanov@gmail.com>
Date: Thu, 17 Mar 2022 17:32:43 +0100
Subject: [PATCH 10/10] Revert "Revert "alt_core.c: Add potentially missing
 assignment of p->on_cpu in sched_fork""

This reverts commit bd3050b86966f0df3eaaf001f1ddd7fc5e228f49.

Signed-off-by: Piotr Gorski <lucjan.lucjanov@gmail.com>
---
 kernel/sched/alt_core.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/kernel/sched/alt_core.c b/kernel/sched/alt_core.c
index 175c9344f..eda09ee3e 100644
--- a/kernel/sched/alt_core.c
+++ b/kernel/sched/alt_core.c
@@ -3056,6 +3056,9 @@ int sched_fork(unsigned long clone_flags, struct task_struct *p)
 #ifdef CONFIG_SCHED_INFO
 	if (unlikely(sched_info_on()))
 		memset(&p->sched_info, 0, sizeof(p->sched_info));
+#endif
+#if defined(CONFIG_SMP)
+	p->on_cpu = 0;
 #endif
 	init_task_preempt_count(p);
 
-- 
2.35.1.354.g715d08a9e5

